### DISCUSSION TITLE
Fails to load built CodeSpace

### DISCUSSION THREAD

**Discussion by mchisolm0** [community/community#34738] [Codespaces]
Students in a team in my class created a CodeSpace from trunk the other day. It was failing the other day, but the CodeSpace seems to be done building now. Consistently, the CodeSpace for both students fails, one with a 403 error and the other student gets the error `No 'CodespaceStateWatcher' found.`.

Here is the students' repo:
https://github.com/Mobile-Apps-Chisolm/collaborative-project-2-fishies

Here is the original repo I gave them for the assignment in GitHub Classroom:
https://github.com/Mobile-Apps-Chisolm/expo-template-js

Please let me know if there is anything else you need from me.

**Comment by jfullstackdev** [comment]
that's 404, 

![image](https://user-images.githubusercontent.com/47092464/193393788-93f1a947-e9ec-470f-9652-27cd465c90ec.png)

while the source, as from your link, is public but it is still a fork

![image](https://user-images.githubusercontent.com/47092464/193393783-3a78a498-15b0-4b90-88b1-cef382dfc400.png)

**Reply by mchisolm0** [reply]
Thank you so much for your answer. I think the reason you got a 404 on the `collaborative-project-2-fishies` is my student repos are set to private. Sorry, I forgot about that. That should be unrelated to my student getting a 404 since that student was signed in and one of the students on that team. Also, I think it is worth noting the student did not get the page you saw for their 404 error. It was during their Codespace creation, so they still had the screen you get when Github tries to build a Codespace from a repo. The title at the top said the build failed, the black output box in the middle showed nothing, and the bottom said "Error details: 404".

**Reply by mchisolm0** [reply]
And I think you are saying the template I am having them use to create their project from is a fork, itself. Is that a problem for creating a Codespace or am I misunderstanding the last sentence and last picture from your reply?

**Comment by mchisolm0** [comment]
I was reading back through [Troubleshooting Prebuilds](https://docs.github.com/en/codespaces/troubleshooting/troubleshooting-prebuilds) and saw it said 

> If the devcontainer.json configuration file for a prebuild configuration is updated to specify that permissions for access to other repositories are required, and a repository administrator has not been prompted to authorize these permissions for the prebuild configuration, then the prebuild workflow may fail.

which took me to [Managing Prebuilds](https://docs.github.com/en/codespaces/managing-your-codespaces/managing-repository-access-for-your-codespaces#setting-additional-repository-permissions) talking about permissions. I did not authorize any new permissions other than the check box in Classroom that makes students admins in the branch they get for their project. I do not think this is the problem, but I still wanted to see if it could be.

**Comment by tryTurningItOffAndBackOn** [comment]
Have you tried turning it off and on again?

**Reply by mchisolm0** [reply]
Try turning what off and on again? The Codespace? The prebuild? The team/students have both tried with different prebuilds and they have consistently failed.
